### PR TITLE
PLT-3150 Added check for code theme name changes

### PR DESCRIPTION
**Pull request by DavidLu1997**
`solaris_dark` and `solaris_light` had their `_`s changed to `-`. This checks for any users with the themes upon server start, and changes them as needed.

Also deleted `remove for 2.4` code

### CODE REVIEW

File: store/sql_user_store.go [hunk 1]
```diff
@@ -47,8 +47,26 @@ func NewSqlUserStore(sqlStore *SqlStore) UserStore {
...
+		for _, user := range data {
+			if user.ThemeProps["codetheme"] == "solarized_dark" {
+				user.ThemeProps["codetheme"] = "solarized-dark"
+				shouldUpdate = true
+			} else if user.ThemeProps["codetheme"] == "solarized_light" {
+				user.ThemeProps["codetheme"] = "solarized-light"
+				shouldUpdate = true
+			}
+
+			if shouldUpdate {
```
Comment (jwilander) [inline store/sql_user_store.go]: You never set `shouldUpdate` back to `false`, so once you hit one user that has the bad theme it's going to update the rest of the users

**Comment by coreyhulen** [comment]
Also lets not remove

```
// ADDED for 2.0 REMOVE for 2.4
-	us.CreateColumnIfNotExists("Users", "Locale", "varchar(5)", "character varying(5)", model.DEFAULT_LOCALE)
```

we remove these in the master branch (it doesn't hurt to leave).  We need to recompute the mappings since version 2.4 doesn't exist and didn't ship.

**Comment by DavidLu1997** [comment]
@jwilander fixed
@coreyhulen put it back in

File: store/sql_user_store.go [hunk 2]
```diff
@@ -49,6 +49,28 @@ func NewSqlUserStore(sqlStore *SqlStore) UserStore {
 func (us SqlUserStore) UpgradeSchemaIfNeeded() {
 	// ADDED for 2.0 REMOVE for 2.4
 	us.CreateColumnIfNotExists("Users", "Locale", "varchar(5)", "character varying(5)", model.DEFAULT_LOCALE)
+
+	// ADDED for 3.2 REMOVE for 3.6
+	var data []*model.User
+	shouldUpdate := false
+	if _, err := us.GetReplica().Select(&data, "SELECT * FROM Users WHERE ThemeProps LIKE '%solaris%'"); err == nil {
```
Comment (hmhealey) [inline store/sql_user_store.go]: The LIKE statement won't match anything because `solarized` doesn't contain `solaris`

**Comment by hmhealey** [comment]
+1

**Comment by crspeller** [comment]
@DavidLu1997 Have you actually tested this?

**Comment by DavidLu1997** [comment]
Admittedly, no. I couldn't replicate it using anything available client side, and I did not know how to edit the database proper. In hindsight, I should have tested it before submitting, and I will definitely thoroughly test everything in the future.

**Comment by DavidLu1997** [comment]
Closing to resubmit

**Pull request closed without merging**